staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -1024,6 +1098,14 @@ func (r *crdHandler) getOrCreateServingInfoFor(uid types.UID, name string) (*crd
 		waitGroup:           &utilwaitgroup.SafeWaitGroup{},
 	}
 
+	currentStorageVersion := "vOther"
+	if previousStorageVersion, ok := r.previousCustomStorage.Load(crd.UID); ok && previousStorageVersion == currentStorageVersion {
+		ret.allowMutation.Store(true)
+	} else {
+		// wait for previous handlers to complete, then store, THEN allow mutation
+		go writeStorageVersion(context.TODO(), ret)
```

Comment:
using a go func here allows the read path to continue, while blocking the write path until the old instances are done serving and the instance has written storageversion.

Reply:
I dont understand why are we trying to update SVs as a part of CR-read handling flow?

Reply:
Also, does this mean we will always be depending on a CR request to happen in order to kick-off a storageversion update? What if no CR writes happen - we wouldnt ever update the storageversion then?